app/models/minion.rb

```diff
@@ -54,6 +63,18 @@ def self.assign_roles!(roles: {}, default_role: :minion)
     end
   end
 
+  def self.select_masters(roles)
```

Comment:
If we want this to scale, we'll absolutely need additional roles - or likely - more flexibility in deployment topology. For example (what follows is a wild guess on numbers) - for >30 nodes, etcd may need to be running on dedicated machines.. for >200 nodes, multiple etcd clusters may be needed, each running on their own machines (for kube core stuff, and the second for kube's event storage). Though, this is a long way off and likely shouldn't be given too much thought just yet.